bindings/experimental/distrdf/python/DistRDF/Proxy.py

```diff
@@ -54,10 +55,28 @@ def execute_graph(node: Node) -> None:
         # Creating a ROOT.TDirectory.TContext in a context manager so that
         # ROOT.gDirectory won't be changed by the event loop execution.
         with _managed_tcontext():
+            # All the information needed to reconstruct the computation graph on
+            # the workers is contained in the head node
             headnode = node.get_head()
+            # Ingredients needed:
+            # - A representation of the computation graph to be executed
+            # - The logical ranges in which the dataset is split
+            # - A callable to build an RDataFrame instance starting from a
+            #   logical range.
             graph_dict = headnode.generate_graph_dict()  # also takes care of pruning the graph
-            generator = ComputationGraphGenerator(headnode, graph_dict)
-            headnode.backend.execute(generator)
+            ranges = headnode.build_ranges()
+            build_rdf_from_range = headnode.generate_rdf_creator()
```

Comment:
You took this out of `execute` and also other things like `handle_returned_values` below, what was the purpose?

Also, what does `build_rdf_from_range` hold?

Reply:
`build_rdf_from_range` is a function that takes in input a logical range and outputs an RDataFrame instance. 
Each distributed task needs to know how to do this, for example starting from a `TreeRange` it needs to create a TChain that reads the correct files (and is restricted to processing only the wanted entries), then the TChain is wrapped by an RDataFrame that will be the starting node of the execution of the computation graph in the task

Reply:
Maybe an alternative for the `BaseBackend.execute` function

```diff
-def execute(self, graph: Dict[int, "Node"], ranges: Iterable, build_rdf_from_range: Callable) -> Iterable:
+def execute(self, headnode: HeadNode) -> Iterable:
```

Taking directly the headnode, no implicit dependency between the classes. Then leaving all the functions of the headnode that were called in the `execute` function could still stay there